src/ngx_http_lua_sleep.c

```diff
@@ -72,10 +72,17 @@ ngx_http_lua_ngx_sleep(lua_State *L)
     coctx->sleep.data = coctx;
     coctx->sleep.log = r->connection->log;
 
-    dd("adding timer with delay %lu ms, r:%.*s", (unsigned long) delay,
-       (int) r->uri.len, r->uri.data);
+    if (delay == 0) {
+        dd("posting 0 sec sleep event to head of delayed queue");
 
-    ngx_add_timer(&coctx->sleep, (ngx_msec_t) delay);
+        coctx->sleep.delayed = 1;
+        ngx_post_event(&coctx->sleep, &ngx_posted_delayed_events);
```

Comment:
Seems like this does not present in older nginx cores like 1.11.2 and 1.9.15? Better protect it with macros?